pkg/apis/sources/v1alpha1/rabbitmq_validation_test.go

```diff
@@ -291,3 +303,64 @@ func TestRabbitmqSourceExchangeConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestRabbitmqSourceSpecValidation(t *testing.T) {
+	testCases := map[string]struct {
+		spec    *RabbitmqSourceSpec
+		wantErr bool
+	}{
+		"valid config": {
+			spec:    &fullSpec,
+			wantErr: false,
+		},
+		"missing rabbitmqClusterReference": {
+			spec:    &RabbitmqSourceSpec{Predeclared: true},
+			wantErr: true,
+		},
+		"missing rabbitmqClusterReference.name": {
+			spec: &RabbitmqSourceSpec{
+				Predeclared:              true,
+				RabbitmqClusterReference: &v1beta1.RabbitmqClusterReference{Namespace: "test"},
+			},
+			wantErr: true,
+		},
+		"missing rabbitmqClusterReference.namespace": {
+			spec: &RabbitmqSourceSpec{
+				Predeclared:              true,
+				RabbitmqClusterReference: &v1beta1.RabbitmqClusterReference{Name: "test"},
+			},
+			wantErr: true,
+		},
+		"including connectionSecret": {
+			spec: &RabbitmqSourceSpec{
+				Predeclared: true,
+				RabbitmqClusterReference: &v1beta1.RabbitmqClusterReference{
+					Namespace: "test", Name: "test", ConnectionSecret: &v1.LocalObjectReference{Name: "test"}},
+			},
+			wantErr: true,
+		},
+		"just connection secret": {
+			spec: &RabbitmqSourceSpec{
+				Predeclared: true,
+				RabbitmqClusterReference: &v1beta1.RabbitmqClusterReference{
+					Namespace: "test", ConnectionSecret: &v1.LocalObjectReference{Name: "test"}},
+			},
+			wantErr: true,
```

Comment:
this should be allowed

Reply:
No, since there are a lot of changes here, the connection secret relies on the broker, username and password params that are been removed, on the next PR that is been solved for the samples to work. that's why I'm modifying these tests too